logstash-core/logstash-core.gemspec

```diff
@@ -1,6 +1,31 @@
 # -*- encoding: utf-8 -*-
 lib = File.expand_path('../lib', __FILE__)
 $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+
+project_versions_yaml_path = File.expand_path("../versions.yml", File.dirname(__FILE__))
+if File.exist?(project_versions_yaml_path)
+  # we need to copy the project level versions.yml into the gem root
+  # to be able to package it into the gems file structure
+  # as the require 'logstash-core/version' loads the yaml file from within the gem root.
+  #
+  # we ignore the copy in git and we overwrite an existing file
+  # each time we build the logstash-core gem
+  original_lines = IO.readlines(project_versions_yaml_path)
+  gem_versions_yaml_path = File.expand_path("./versions-gem-copy.yml", File.dirname(__FILE__))
+  version_difference = true
+  if File.exists?(gem_versions_yaml_path)
+    copy_lines = IO.readlines(project_versions_yaml_path)
+    version_difference = original_lines[2] != copy_lines[2]
```

Comment:
Can you add a comment to what this is specifically checking?

Reply:
Actually can we just get rid of this, and always ensure a delete->copy -> comment ? E.g. Remove the version_diff conditional?

Reply:
Will do (it was so but I changed it because the gemspec gets evaluated soooo many times during various rake or bundler activities).

Reply:
in that case would md5 hash of the file be more appropriate ?

Reply:
No need, I said I would revert to overwrite (its about the same computation cost to hashing anyway)
Actually, there is a flaw in the current check because we may only be changing the jrjackson and jackson values but then the file would not be copied (say, during development == POLS violation)
It was not an on-purpose thing to check if the reviewer brought their A game, I'm not Machiavelli. 😆